test_functions.h

```diff
@@ -0,0 +1,37 @@
+#ifndef TEST_FUNCTIONS_H
+#define TEST_FUNCTIONS_H
+
+#define ASSERT_EQ(expected, actual) do { \
+if ((expected) != (actual)) { \
+        printf("TEST FAILED!!! Bad cred in ovector: expected %d, actual %d but we continue\n", (expected), (actual)); \
```

Comment:
make this messages more generic

For example: "TEST FAILED! actual : actual %d != expected %d"